compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaIntersectionTypeSymbol.java

```diff
@@ -73,18 +74,25 @@ public TypeSymbol effectiveTypeDescriptor() {
         TypesFactory typesFactory = TypesFactory.getInstance(this.context);
         BType effectiveType = ((BIntersectionType) this.getBType()).effectiveType;
         this.effectiveType = typesFactory.getTypeDescriptor(effectiveType,
-                                                            effectiveType != null ? effectiveType.tsymbol : null,
-                                                            false, false, true);
+                effectiveType != null ? effectiveType.tsymbol : null,
+                false, false, true);
         return this.effectiveType;
     }
 
     @Override
     public List<FunctionSymbol> langLibMethods() {
         if (this.langLibFunctions == null) {
-            LangLibrary langLibrary = LangLibrary.getInstance(this.context);
-            List<FunctionSymbol> functions = langLibrary.getMethods(
-                    ((AbstractTypeSymbol) this.effectiveTypeDescriptor()).getBType());
-            this.langLibFunctions = filterLangLibMethods(functions, this.getBType());
+            if (this.effectiveTypeDescriptor().typeKind() == TypeDescKind.OBJECT) {
+                this.langLibFunctions = this.effectiveTypeDescriptor().langLibMethods();
+            } else if (this.effectiveTypeDescriptor().typeKind() == TypeDescKind.TYPE_REFERENCE) {
+                TypeReferenceTypeSymbol typeRef = (TypeReferenceTypeSymbol) this.effectiveTypeDescriptor();
```

Comment:
This wouldn't work if it's a deeply nested type reference right? Shall we make it a recursive call to unwrap until its no longer a type ref?

Reply:
In here the effectiveTypeDescriptor method gives the correct result. We are using that type symbol to get the langlibs. So this is not an issue right?

```
type Type1 object {

};

type Type2 readonly & Type1;

type Type3 Type2;

type Type4 Type3;

function name() {
  Type4 t =  object {};
  t.<cursor>
}
```
I tried with the above example.

Reply:
We are creating type symbol for each reference. But if it works you might be calling it recursively in an underlying method. Add tests as such which has more 2 references

Reply:
It works fine.

Reply:
Commit those tests

Reply:
Added [166d2dce2e0d3f10338a2bd4b976c5894c4ed58d](https://github.com/ballerina-platform/ballerina-lang/pull/39520/commits/166d2dce2e0d3f10338a2bd4b976c5894c4ed58d)